rse-ssh.Rmd

```diff
@@ -0,0 +1,449 @@
+# Working Remotely {#rse-ssh}
+
+```{r rse-ssh-setup, include=FALSE}
+source("_common.R")
+```
+
+Let's take a closer look at what happens when we use the shell
+on a desktop or laptop computer.
+The first step is to log in
+so that the operating system knows who we are and what we're allowed to do.
+We do this by typing our username and password;
+the operating system checks those values against its records,
+and if they match,
+runs a shell for us.
+
+As we type commands,
+the 1's and 0's that represent the characters we're typing are sent from the keyboard to the shell.
+The shell displays those characters on the screen to represent what we type,
+and then,
+if what we typed was a command,
+the shell executes it and displays its output (if any).
+
+What if we want to run some commands on another machine,
+such as the server in the basement that manages our database of experimental results?
+To do this,
+we have to first log in to that machine.
+We call this a [remote login][remote-login].
+
+In order for us to be able to login, the remote computer must be running 
+a [remote login server][remote-login-server] and we will
+run a client program that can talk to that server.
+The client program passes our login credentials to the remote login server
+and, if we are allowed to login, that server then runs a shell for us on the 
+remote computer.
+
+Once our local client is connected to the remote server,
+everything we type into the client is passed on, by the server, to the shell 
+running on the remote computer.
+That remote shell runs those commands on our behalf,
+just as a local shell would,
+then sends back output, via the server, to our client, for our computer to display.
+
+## How did we get here? {#rse-ssh-history}
```

Comment:
```suggestion
```

I don't think this needs to be its own section.

Reply:
Removed - thanks.